src/hash.hpp

```diff
@@ -24,6 +26,40 @@ namespace Simhash {
      * @return hash representative of the content of the text */
     template <typename Hash=jenkins>
     class Simhash {
+    private:
+        class Accumulator {
+        private:
+            int64_t v[64];
+
+            size_t window;
+            Cyclic<hash_t> cyclic;
+
+        public:
+            Accumulator(): window(3), cyclic(window) {
+                memset(v, 0, 64);
+            }
+
+            void update(hash_t hash) {
+                int j(0);
+                for (j = 63; j > 0; --j) {
```

Comment:
Refactored. No reason for it to be as it is now. The reason it was `j > 0` in its previous incarnation is that it was using an unsigned variable.